codecarbon/core/cpu.py

```diff
@@ -209,3 +211,36 @@ def get_cpu_details(self) -> Dict:
                 exc_info=True,
             )
         return cpu_details
+
+
+class TDP:
+    def __init__(self):
+        self.tdp = self._get_power_from_constant()
+
+    def _parse_cpu_model(self, raw_name) -> str:
+        """
+        Parse the model name from the raw name extracted from cpuinfo library
+        :return: parsed CPU name
+        """
+        return (
+            raw_name.split(" @")[0]
+            .replace("(R)", "")
+            .replace("(TM)", "")
+            .replace(" CPU", "")
+        )
+
+    def _get_power_from_constant(self) -> int:
+        """
+        Get CPU power from constant mode
+        :return: power in Watt
+        """
+        cpu_info = cpuinfo.get_cpu_info()
+        if cpu_info:
+            model_raw = cpu_info["brand_raw"]
+            model = self._parse_cpu_model(model_raw)
```

Comment:
This can fails as we don't have any guarantee on the format of `model_raw`. We need to catch the exception to avoid crashing the user script and decide what to do in that case.

Reply:
I don't think the parsing function would fail even if the format is not guaranteed. The parsing is basically just split and replace - it would do nothing for that step if any of those patterns are not detected.

Reply:
So far, assuming that `model_raw` is always a string, I don't see a way for it to fails. We can tackle it later if you prefer.

Reply:
I'll add an exception in case in the future `cpuinfo` library will return something other than string (or even they changed the key name for "brand raw"). But right now it seems that it will always return a string.
https://github.com/workhorsy/py-cpuinfo